streams/src/main/java/org/apache/kafka/streams/query/KeyQuery.java

```diff
@@ -40,7 +42,16 @@ private KeyQuery(final K key) {
      * @param <V> The type of the value that will be retrieved
      */
     public static <K, V> KeyQuery<K, V> withKey(final K key) {
-        return new KeyQuery<>(key);
+        return new KeyQuery<>(key, false);
+    }
+
+    /**
+     * Specifies that the cache should be skipped during query evaluation. This means, that the query will always
+     * get forwarded to the underlying store.
+     */
+    @SuppressWarnings("unchecked")
+    public <K, V> KeyQuery<K, V> skipCache() {
+        return new KeyQuery<>((K) key, true);
```

Comment:
```suggestion
    public KeyQuery<K, V> skipCache() {
        return new KeyQuery<>(key, true);
```

I think you won't need the cast or the suppression if you just keep the K and V bound to the same values as the current instance (`this`).